spec/models/git_repository_spec.rb

```diff
@@ -0,0 +1,163 @@
+require 'spec_helper'
+
+describe GitRepository do
+  it "no url" do
+    expect { FactoryGirl.create(:git_repository) }.to raise_error(ActiveRecord::RecordInvalid)
+  end
+
+  it "invalid url" do
+    expect { FactoryGirl.create(:git_repository, :url => "abc") }.to raise_error(ActiveRecord::RecordInvalid)
+  end
+
+  it "default dirname" do
+    repo = FactoryGirl.create(:git_repository,
+                              :url => "http://www.something.com/repos/manageiq")
```

Comment:
If I recall, fake URLs should use example.com...something.com may be a real location on the web, so we don't want to hit it accidentally.